src/recon.rs

```diff
@@ -742,7 +742,11 @@ fn decode_coefs<BD: BitDepth>(
                 };
                 let shift2: c_uint = 0;
                 let mask: c_uint = 4 * sh as c_uint - 1;
-                levels[..stride * (4 * sw as usize + 2)].fill(0);
+                // Optimizes better than `.fill(0)`,
+                // which doesn't elide the bounds check, inline, or vectorize.
+                for i in 0..stride * (4 * sw as usize + 2) {
+                    levels[i] = 0;
+                }
```

Comment:
Yeah, I was looking at the assembly.  I think it's because `.fill` just calls `memset` instead of taking advantage of extra information known about the length.